core/config.go

```diff
@@ -85,6 +86,15 @@ func (d *Config) ListenAddress(defaultAddr string) string {
 	return defaultAddr
 }
 
+// ControlPort returns the default port for control port communications (8888)
+// or the port stored in the config thanks to WithControlPort
+func (d *Config) ControlPort() string {
+	if d.controlPort != "" {
```

Comment:
You can set the default directly in the `NewConfig` method, so (1) you don't have to do this check each time, and (2) all constants are set up in one and only one place :)